flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java

```diff
@@ -148,6 +148,30 @@ public class BatchExecutionOptions {
                     .withDescription(
                             "Controls how long an detected slow node should be blocked for.");
 
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Boolean> JOB_RECOVERY_ENABLED =
+            key("execution.batch.job-recovery.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("A flag to enable or disable the job recovery.");
+
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Duration> JOB_RECOVERY_PREVIOUS_WORKER_RECOVERY_TIMEOUT =
+            key("execution.batch.job-recovery.previous-worker.recovery.timeout")
+                    .durationType()
+                    .defaultValue(Duration.ofSeconds(30))
+                    .withDescription(
+                            "The timeout for a new job master to wait for the previous worker to reconnect.");
```

Comment:
More explanations: "A reconnected worker will transmit the details of its produced intermediate results to the new job master, enabling the job master to reuse these results."